sharing/filter_single_value.py

```diff
@@ -0,0 +1,137 @@
+"""
+This module filters data by a single rule value of any datatype.
+"""
+
+from typing import Tuple, List, Union  # pylint: disable=import-error
+import re
+import pandas as pd  # pylint: disable=import-error
+from pandas.core.frame import DataFrame  # pylint: disable=import-error
+from check_rule_type import check_type_of_rule
+
+
+def filter_single_value(
```

Comment:
The logic of dropping columns and rows from a dataset is duplicated inside this and the filter_range_values function. These functions should really be about checking whether a set of row(s) and column(s) should be excluded from a dataset based on a rule value. The actual filtering logic should be in the loop_through_tables function.